### ISSUE TITLE
How to update root directory of nvim-tree upon `:cd`?

### ISSUE THREAD

**Issue by winston0410** [kyazdani42/nvim-tree.lua#356]
How can I update the root directory of nvim-tree upon `:cd`? In nerdTree, this is what I use:

```vim
"Plug 'preservim/nerdtree' | 
  "\ Plug 'Xuyuanp/nerdtree-git-plugin'
let NERDTreeShowHidden=1
let g:NERDTreeWinPos = "left"
let NERDTreeMinimalUI=1

"nerdtree-git-plugin"
let g:NERDTreeGitStatusUseNerdFonts = 1
let g:NERDTreeGitStatusIndicatorMapCustom = {
                \ 'Modified'  :'',
                \ 'Staged'    :'✚',
                \ 'Untracked' :'✭',
                \ 'Renamed'   :'➜',
                \ 'Unmerged'  :'═',
                \ 'Deleted'   :'✖',
                \ 'Dirty'     :'✗',
                \ 'Ignored'   :'☒',
                \ 'Clean'     :'✔︎',
                \ 'Unknown'   :'?',
                \ }

function RefreshNERDTree()
    if g:NERDTree.IsOpen()
        :NERDTreeRefreshRoot
        :NERDTreeCWD
    endif
endfunction

function NERDTreeToggleAndRefresh()
  :NERDTreeToggle
  :call RefreshNERDTree()
endfunction

augroup DIRCHANGE
    au!
    autocmd DirChanged tabpage silent :call RefreshNERDTree()
augroup END
```

Is there any similar command like `:NERDTreeCWD` in `nvim-tree`?

**Comment by sindrets** [comment]
It can be implemented by using NvimTree's API:

```lua
local M = {}

function M.update_cwd()
  local view = require'nvim-tree.view'
  local lib = require'nvim-tree.lib'
  if view.win_open() then
    lib.change_dir(vim.fn.getcwd())
  end
end

vim.api.nvim_exec([[
  augroup NvimTreeConfig
    au!
    au DirChanged * lua NvimTreeConfig.update_cwd()
  augroup END
  ]], false)

_G.NvimTreeConfig = M
return M
```

**Comment by kyazdani42** [comment]
@sindrets i think we should make that an nvim-tree internal. it was not possible before but i guess now it works with your solution :)

**Comment by kyazdani42** [comment]
this is now fixed in  3196205131a2d4
